### PR TITLE
zigbee: Fix high & medium impact coverity issues

### PR DESCRIPTION
**Pull request by maciekfabia**
Fix coverity issues having high or medium impact
in zigbee.

### CODE REVIEW

**Review by sebastiandraus — APPROVED** [review]
LGTM

File: samples/zigbee/light_bulb/src/main.c [hunk 1]
```diff
@@ -497,7 +497,7 @@ void error(void)
 	}
 }
 
-void main(void)
+int main(void)
```
Comment (rlubos) [inline samples/zigbee/light_bulb/src/main.c:500]: Zephyr defines main as `void main(void);` (https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/init.c#L263) and all of the samples seem to follow this pattern, I'm not sure if we should change it in ZB (and what consequences would that have).

Comment (maciekfabia) [reply samples/zigbee/light_bulb/src/main.c]: @rlubos you're right, besides `main` is just a low impact issue reported by coverity.
I changed it back to `void main(void)`.

**Review by rlubos — APPROVED** [review]

**Comment by maciekfabia** [comment]
Rebased.
@rlubos can we merge?

**Comment by rlubos** [comment]
Sure

**Pull request merged**